epaint/src/color.rs

```diff
@@ -74,6 +74,19 @@ impl Color32 {
         Self([r, g, b, 255])
     }
 
+    #[inline(always)]
+    pub fn from_hex(hex: &str) -> Result<Self, ParseIntError> {
+        let [r, g, b, _] = u32::from_str_radix(&hex.replace("#", ""), 16)?.to_be_bytes();
+        Ok(Self([r, g, b, 255]))
```

Comment:
I'd like for these functions to only accept `#rrggbb` and `#rrggbbaa`, and NOT e.g. `#123`. Also they need tests.

Reply:
like ensure that the hex is valid

```rust
pub fn validate_hex_string(hex: &str) -> Result<(), ()> {
    match &hex.len() {
        x if *x < 7 => return Err(()), // Truncated
        x if *x > 7 => return Err(()), // Overflow
        _ => {}
    };
    if &hex[0..1] != "#" {
        return Err(()); // Invalid
    }
    Ok(())
}
```